### PR TITLE
V3: Graph shows selection plus refactoring to use dataset as ultimate source

### PR DESCRIPTION
**Pull request by bfinzer**
* Point selection (click or marquee) selects cases in dataset
* Point selection in graph stays in synch with case selection in dataset
* Values that determine point location are derived from attribute values in dataset
* Circles are removed when a new dataset is discovered rather than left in DOM

### CODE REVIEW

File: v3/src/components/graph/dotplotdots.tsx
Comment (kswenson) [inline v3/src/components/graph/dotplotdots.tsx:13]: Move import ⬆️

File: v3/src/components/graph/dotplotdots.tsx
Comment (kswenson) [inline v3/src/components/graph/dotplotdots.tsx:36]: FYI, this could also be:
```
    selectedDataObjects = useRef<{Record<string, { x: number }>}>({}),
```

File: v3/src/components/graph/dotplotdots.tsx
Comment (kswenson) [inline v3/src/components/graph/dotplotdots.tsx:56]: It probably shouldn't ever happen, but I would think that defaulting to `NaN` would make more sense than `0`. Otherwise, it would run the risk of being plotted as data.

File: v3/src/components/graph/dotplotdots.tsx
Comment (kswenson) [inline v3/src/components/graph/dotplotdots.tsx:64]: Could also just be:
```
      if (dragID) {
```

File: v3/src/components/graph/scatterdots.tsx
Comment (kswenson) [inline v3/src/components/graph/scatterdots.tsx:70]: Rather than running the risk of changing `NaN`s to numbers in the data, I'd've thought something like this would make more sense:
```
            if (isFinite(currX) && isFinite(currY)) {
              worldDataRef.current?.setValue(anID, xAttrID, Number(currX) + deltaX)
              worldDataRef.current?.setValue(anID, yAttrID, Number(currY) + deltaY)
            }
```
or even:
```
            if (isFinite(currX) && isFinite(currY)) {
              worldDataRef.current?.setCaseValues([{
                __id__: anID, [xAttrID]: currX + deltaX, [yAttrID]: currY + deltaY
              }])
            }
```
which has the advantage of being a single action at the model layer.

File: v3/src/components/graph/scatterdots.tsx
Comment (kswenson) [inline v3/src/components/graph/scatterdots.tsx:88]: As described in a comment above, this could be replaced with a single call to `setCaseValues()`.

File: v3/src/data-model/data-set.ts
Comment (kswenson) [inline v3/src/data-model/data-set.ts:593]: Let's leave this for now but I've started working on the refactor we discussed which will change this (e.g. `setCaseValue()`).

**Review by kswenson — CHANGES_REQUESTED** [review]
👍  looks good! I made a bunch of suggestions but most of them are points of interest and I leave it to your discretion whether to address them now or not. The one that really should be fixed before the PR goes in is the conditional calling of the `useGetData()` hook, which oddly isn't triggering a rules-of-hooks warning. I also found a few bugs that you may or may not want to address before the PR is merged:

- after initial switch to scatter plot, Y axis isn't draggable and label doesn't appear. Everything works after a redraw is triggered, e.g. by dragging the X axis.
- movable line isn't visible initially with mammals data (height on Y axis so scale only goes to 6.5 but movable line intercept defaults to 10).
- dots are visible below X axis when translating the Y axis (clipping issue).

File: v3/src/components/graph/graph-hooks/graph-hooks.ts
Comment (bfinzer) [reply v3/src/components/graph/graph-hooks/graph-hooks.ts:36]: Let's talk about how this works.

**Pull request merged**
